modules/navier_stokes/include/fvkernels/INSFVElementalKernel.h

```diff
@@ -26,23 +26,30 @@ class INSFVElementalKernel : public FVElementalKernel, public INSFVMomentumResid
 
   virtual ~INSFVElementalKernel() = default;
 
-  void computeResidual() override final {}
-  void computeJacobian() override final {}
+  void computeResidual() override;
+  void computeJacobian() override;
   using FVElementalKernel::computeOffDiagJacobian;
   void computeOffDiagJacobian() override final {}
-  void computeResidualAndJacobian() override final {}
+  void computeResidualAndJacobian() override;
 
 protected:
   ADReal computeQpResidual() override final
   {
     mooseError("INSFVElementalKernels must implement gatherRCData and not computeQpResidual");
   }
 
+  /// Compute the contribution which goes into the residual of the segregated system. This
+  /// needs to accomodate the different linearization approaches needed to get the suitable
+  /// system matrix contributions when the Jacobian assembly routine is called.
+  virtual ADReal computeSegregatedContribution()
+  {
+    mooseError(
+        this->type(),
+        " needs to implement computeSegregatedContribution to be usable with a segregated solver!");
+  }
+
   /**
    * Process into either the system residual or Jacobian
    */
   void addResidualAndJacobian(const ADReal & residual, dof_id_type dof);
-
-private:
-  using FVElementalKernel::_current_elem;
```

Comment:
Hmm

Reply:
I need these for the `computeSegregatedContribution()` functions.

Reply:
I was thinking about setting it in the gatherRC loop too so that it is consistent, but abandoned the idea

Reply:
I remember I did this because I spent hours tracking down a bug where I was using an invalid `_current_elem` (which is only correctly set when doing the framework level residual/Jacobian computation ... it is not set during the RC data gathering loop) in a derived class. We can leave this as is since you needed it. Perhaps we'll do something more in the future if I shoot myself in the foot again 😄